osu.Game/Overlays/Mods/ModButton.cs

```diff
@@ -64,41 +64,45 @@ private bool changeSelectedIndex(int newIndex)
             if (newIndex >= 0 && !Mods[newIndex].HasImplementation)
                 return false;
 
-            selectedIndex = newIndex;
-            Mod modAfter = SelectedMod ?? Mods[0];
-
-            if (beforeSelected != Selected)
+            Schedule(() =>
             {
-                iconsContainer.RotateTo(Selected ? 5f : 0f, 300, Easing.OutElastic);
-                iconsContainer.ScaleTo(Selected ? 1.1f : 1f, 300, Easing.OutElastic);
-            }
+                selectedIndex = newIndex;
+                Mod modAfter = SelectedMod ?? Mods[0];
 
-            if (modBefore != modAfter)
-            {
-                const float rotate_angle = 16;
+                if (beforeSelected != Selected)
+                {
+                    iconsContainer.RotateTo(Selected ? 5f : 0f, 300, Easing.OutElastic);
+                    iconsContainer.ScaleTo(Selected ? 1.1f : 1f, 300, Easing.OutElastic);
+                }
+
+                if (modBefore != modAfter)
+                {
+                    const float rotate_angle = 16;
 
-                foregroundIcon.RotateTo(rotate_angle * direction, mod_switch_duration, mod_switch_easing);
-                backgroundIcon.RotateTo(-rotate_angle * direction, mod_switch_duration, mod_switch_easing);
+                    foregroundIcon.RotateTo(rotate_angle * direction, mod_switch_duration, mod_switch_easing);
+                    backgroundIcon.RotateTo(-rotate_angle * direction, mod_switch_duration, mod_switch_easing);
 
-                backgroundIcon.Mod = modAfter;
+                    backgroundIcon.Mod = modAfter;
 
-                using (BeginDelayedSequence(mod_switch_duration, true))
-                {
-                    foregroundIcon
-                        .RotateTo(-rotate_angle * direction)
-                        .RotateTo(0f, mod_switch_duration, mod_switch_easing);
+                    using (BeginDelayedSequence(mod_switch_duration, true))
+                    {
+                        foregroundIcon
+                            .RotateTo(-rotate_angle * direction)
+                            .RotateTo(0f, mod_switch_duration, mod_switch_easing);
 
-                    backgroundIcon
-                        .RotateTo(rotate_angle * direction)
-                        .RotateTo(0f, mod_switch_duration, mod_switch_easing);
+                        backgroundIcon
+                            .RotateTo(rotate_angle * direction)
+                            .RotateTo(0f, mod_switch_duration, mod_switch_easing);
 
-                    Schedule(() => displayMod(modAfter));
+                        Schedule(() => displayMod(modAfter));
+                    }
                 }
-            }
 
-            foregroundIcon.Selected.Value = Selected;
+                foregroundIcon.Selected.Value = Selected;
+
+                SelectionChanged?.Invoke(SelectedMod);
```

Comment:
Unfortunately this is not all too safe to schedule because of the <kbd>Ctrl</kbd>+<kbd>Enter</kbd> shortcut. Try the following:

1. Start gameplay with autoplay via <kbd>Ctrl</kbd>+<kbd>Enter</kbd>
2. Quit gameplay after a bit
3. Open the mod overlay
4. The autoplay mod button will select again

A similar thing will happen if the mod overlay is opened at time of pressing the hotkey (it'll get selected fine, but then play the deselect song effect at step 4).

Reply:
i think that should be safe to move out, actually. probably shouldn't have been in there..

Reply:
I've moved the pieces out which affect the event / selection state flow (so the scheduled part specifically covers transforms only).

Not too hot on this code overall - it's very unlikely all these variables and checks are required to update the animated state of a single button. But that's part of a larger problem (code quality of the mod overlay in general is poor).